src/DIPS.Xamarin.UI/Controls/Toast/Toast.cs

```diff
@@ -0,0 +1,274 @@
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace DIPS.Xamarin.UI.Controls.Toast
+{
+    /// <summary>
+    ///     Toast control that would appear on top of the presented view
+    /// </summary>
+    public class Toast
+    {
+        private Toast()
+        {
+            Application.Current.PageAppearing -= OnPageAppearing;
+            Application.Current.PageAppearing += OnPageAppearing;
+
+            Application.Current.PageDisappearing -= OnPageDisappearing;
+            Application.Current.PageDisappearing += OnPageDisappearing;
+        }
+
+        /// <summary>
+        ///     Get the current instance of the Toast control
+        /// </summary>
+        public static Toast Current { get; } = new Toast();
+
+        private CancellationTokenSource CancellationSource { get; set; } = new CancellationTokenSource();
+        private Dictionary<string, Grid> ToastContainers { get; } = new Dictionary<string, Grid>();
+        private ToastLayout ToastLayout { get; set; }
+        private ToastOptions ToastOptions { get; set; }
+
+        private void OnPageDisappearing(object sender, Page page)
+        {
+            if (page is ContentPage contentPage)
+            {
+                CancellationSource.Cancel();
+                _ = HideToast(contentPage);
+            }
+        }
+
+        private void OnPageAppearing(object sender, Page e)
+        {
+            Initialize();
+        }
+
+        /// <summary>
+        ///     Initialize Toast control.
+        /// </summary>
+        public void Initialize()
+        {
+            _ = GetToastContainerSettingUpIfNeededAsync();
+        }
+
+        /// <summary>
+        ///     Displays the Toast control
+        /// </summary>
+        /// <param name="text">Text to be displayed in the Toast control</param>
+        /// <param name="options">An <see cref="Action{ToastOptions}" /> to modify Toast options</param>
+        /// <param name="layout">An <see cref="Action{ToastLayout}" /> to modify Toast layout</param>
+        /// <returns>A void <c>Task</c></returns>
+        public async Task DisplayToast(string text, Action<ToastOptions> options, Action<ToastLayout> layout)
+        {
+            var toastOptions = new ToastOptions();
+            options(toastOptions);
+            var layoutOptions = new ToastLayout();
+            layout(layoutOptions);
+
+            await DisplayToast(text, toastOptions, layoutOptions);
+        }
+
+        /// <summary>
+        ///     Displays the Toast control
+        /// </summary>
+        /// <param name="text">Text to be displayed in the Toast control</param>
+        /// <param name="options"><see cref="ToastOptions" /> to set for the Toast control</param>
+        /// <param name="layout"><see cref="ToastLayout" /> to set for the Toast control</param>
+        /// <returns>A void <c>Task</c></returns>
+        public async Task DisplayToast(string text, ToastOptions options = null, ToastLayout layout = null)
```

Comment:
Should we catch exceptions of the API?

Reply:
Any specific location of concern?

Reply:
Not really, just in case.

Reply:
wrapped the API methods with `try` `catch`